Assets/Scripts/AppController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.UI;
+
+/// <summary>
+/// AppController is used to control the length of the experience and display how long until the game ends. It also tells <see cref="Fire"/> that users can fire/not fire, <see cref="SpawnSystem"/> to stop spawning, and <see cref="Reticule"/> to hide the crosshairs from view.
+/// </summary>
+public class AppController
+    : MonoBehaviour
+{
+    public float MaxGameLength;
+    public float CurrentTime { get; private set; }
```

Comment:
Separate Variables, Properties and Methods.

Reply:
Done